# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ImageContainer.cpp
# Commit: 35d86243cb45
# Full Hash: 35d86243cb45f7e02dd6ec0eea04cd62c2a70196
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-09-16 09:14:26
# Regressor Bug: 1908245
# File Overlap Count: 1
# Description:
#   Bug 1908245 - Add support for missing requestVideoFrameCallback parameters. r=media-playback-reviewers,webidl,padenot,saschanaz
#   
#   This patch adds support for exposing the capture time, receive time,
#   processing duration and RTP timestamp parameters on the callback for
#   rVFC for WebRTC related elements.
# ==============================================================================

diff -r 7d143d92bf2a -r 35d86243cb45 gfx/layers/ImageContainer.cpp
--- a/gfx/layers/ImageContainer.cpp	Sun Sep 15 20:03:40 2024 +0000
+++ b/gfx/layers/ImageContainer.cpp	Sun Sep 15 21:49:38 2024 +0000
@@ -329,6 +329,11 @@
     OwningImage* img = newImages.AppendElement();
     img->mImage = aImages[i].mImage;
     img->mTimeStamp = aImages[i].mTimeStamp;
+    img->mProcessingDuration = aImages[i].mProcessingDuration;
+    img->mMediaTime = aImages[i].mMediaTime;
+    img->mWebrtcCaptureTime = aImages[i].mWebrtcCaptureTime;
+    img->mWebrtcReceiveTime = aImages[i].mWebrtcReceiveTime;
+    img->mRtpTimestamp = aImages[i].mRtpTimestamp;
     img->mFrameID = aImages[i].mFrameID;
     img->mProducerID = aImages[i].mProducerID;
     for (const auto& oldImg : mCurrentImages) {